### PR TITLE
Develop a way for RNode operators to learn about the bond status of a node

### PR DESCRIPTION
**Pull request by itegulov**
## Overview
Adds a command `rnode bond-status <publickey>` for checking whether a given validator is bonded or not in the last finalized block.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3066

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/api/BondedStatusAPITest.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+  }
+
+  "bondStatus" should "return false for not bonded validators" in effectTest {
+    TestNode.standaloneEff(genesisContext).use { node =>
+      val (_, publicKey) = Secp256k1.newKeyPair
+      bondedStatus(publicKey)(node) shouldBeF false
+    }
+  }
+}
```
Comment (odeac) [inline casper/src/test/scala/coop/rchain/casper/api/BondedStatusAPITest.scala:86]: We are missing a test showing that bondStatus returns false for a validator that is bonded in a block which is not yet finalized

**Review by odeac — CHANGES_REQUESTED** [review]

**Comment by itegulov** [comment]
@odeac PTAL

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala [hunk 2]
```diff
@@ -635,4 +635,24 @@ object BlockAPI {
       )
     )
   }
+
+  def bondStatus[F[_]: Monad: EngineCell: Log](
+      publicKey: ByteString
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala:639]: Is this a private key, public key, validator identity, etc? I believe we have specific types for each one.

**Review by KentShikama — APPROVED** [review]

**Review by odeac — APPROVED** [review]

**Pull request merged**